src/Slices/Notification/UI/Drawer/Drawer.tsx

```diff
@@ -16,31 +11,53 @@ import {
   NotificationDrawerList,
 } from "@patternfly/react-core";
 import { EllipsisVIcon } from "@patternfly/react-icons";
-import { RemoteData } from "@/Core";
-import { DependencyContext } from "@/UI/Dependency";
+import { UseMutateFunction, UseQueryResult } from "@tanstack/react-query";
+import { PageSize } from "@/Core";
+import {
+  NotificationResponse,
+  useGetNotifications,
+} from "@/Data/Managers/V2/Notification/GetNotifications";
+import {
+  UpdateNotificationParams,
+  useUpdateNotification,
+} from "@/Data/Managers/V2/Notification/UpdateNotification";
 import { useNavigateTo } from "@/UI/Routing";
 import { words } from "@/UI/words";
-import { Body } from "@S/Notification/Core/Domain";
-import { drawerQuery, ViewData } from "@S/Notification/Core/Query";
 import { Item, OnUpdate } from "./Item";
 
+/**
+ * Props for the Drawer component.
+ *
+ * @property {(event?: MouseEvent) => void} onClose - Function to handle drawer closing
+ * @property {boolean} isDrawerOpen - Whether the drawer is currently open
+ * @property {MutableRefObject<HTMLDivElement | undefined>} drawerRef - Ref to the drawer DOM element
+ */
 interface Props {
   onClose(event?: MouseEvent): void;
   isDrawerOpen: boolean;
   drawerRef: MutableRefObject<HTMLDivElement | undefined>;
 }
 
+/**
+ * Main notification drawer component that displays a list of notifications.
+ * Provides functionality to view, mark as read/unread, and clear notifications.
+ */
 export const Drawer: React.FC<Props> = ({
   onClose,
   isDrawerOpen,
   drawerRef,
 }) => {
-  const { commandResolver, queryResolver } = useContext(DependencyContext);
-  const data = queryResolver.useReadOnly<"GetNotifications">(drawerQuery);
-
-  const trigger = commandResolver.useGetTrigger<"UpdateNotification">({
-    kind: "UpdateNotification",
+  const response = useGetNotifications({
+    pageSize: PageSize.from("250"),
```

Comment:
Hmm do we really want to have a list of 250 notifications in the drawer? Seems like overkill.